src/storage/mod.rs

```diff
@@ -480,15 +476,12 @@ impl Options {
     }
 }
 
-struct StorageHandle {
-    handle: Option<thread::JoinHandle<()>>,
-    receiver: Option<Receiver<Msg>>,
-}
-
 pub struct Storage {
     engine: Box<Engine>,
-    sendch: SyncSendCh<Msg>,
-    handle: Arc<Mutex<StorageHandle>>,
+
+    // to schedule the execution of storage commands
+    worker: Arc<Mutex<Worker<Msg>>>,
```

Comment:
do we need `Arc<Mutex>` here?

Reply:
yes, if we do not change `Worker`.

Reply:
I discussed with @hicqu and we think maybe moving `Worker` in an upper level, you can checkout the latest commit. There are a few things you should pay attention to:

1. Storage can't be simply be used like this:
```
let config = Config::default();
let mut storage = Storage::new(&config).unwrap();
storage.start(&config).unwrap();

storage.stop();
```

it has to be like this:
```
let mut worker = Worker::new(...);
let mut storage = Storage::new(&config, worker.scheduler()).unwrap();
storage.start(&config, &worker);

storage.stop(&worker)
```
2. the start of storage is moved earlier when TiKV start

If this is method is ok, I will fix up the unit test and other api issue.